File: src/main/java/pl/yalgrin/playnite/simplesync/repository/CompletionStatusRepository.java
package pl.yalgrin.playnite.simplesync.repository;

import org.springframework.data.r2dbc.repository.Query;
import org.springframework.stereotype.Repository;
import pl.yalgrin.playnite.simplesync.domain.CompletionStatus;
import reactor.core.publisher.Flux;

@Repository
public interface CompletionStatusRepository extends ObjectRepository<CompletionStatus> {
    @Query("select c.id from playnite_comp_status c order by c.id asc")
    Flux<Long> findAllIds();
}

File: src/main/kotlin/pl/yalgrin/playnite/simplesync/library/repository/CompletionStatusRepository.kt
package pl.yalgrin.playnite.simplesync.library.repository

import org.springframework.data.r2dbc.repository.Query
import org.springframework.stereotype.Repository
import pl.yalgrin.playnite.simplesync.library.domain.CompletionStatus
import reactor.core.publisher.Flux

@Repository
interface CompletionStatusRepository : ObjectRepository<CompletionStatus> {
    @Query("select c.id from playnite_comp_status c order by c.id")
    override fun findAllIds(): Flux<Long>
}
